### PR TITLE
Require explicit flag to calculate available_models for search

### PR DESCRIPTION
**Pull request by crisptrutski**
It's a lot of work to calculate the `available_models`, and in most cases we do not use this data. This PR makes the calculation opt-in.

TODO 
- [x] Send through the new flag where appropriate from the frontend

### CODE REVIEW

**Comment by crisptrutski** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/metabase/metabase/48133?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#48133** <a href="https://app.graphite.dev/github/pr/metabase/metabase/48133?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#48108** <a href="https://app.graphite.dev/github/pr/metabase/metabase/48108?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @crisptrutski and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: src/metabase/search/impl.clj
Comment (crisptrutski) [inline src/metabase/search/impl.clj:590]: These were very nearly alphabetically sorted, so I resorted them.

**Review by sansolovyov — COMMENTED** [review]
It contains traces of previous PR though

**Review by sansolovyov — APPROVED** [review]

**Comment by crisptrutski** [comment]
> It contains traces of previous PR though

Wish Github was a bit smarter about squashing. Fixed.

**Comment by crisptrutski** [comment]
Test failures are just some tests needing to be updated.

**Comment by crisptrutski** [comment]
@npfitz Is this g2g?

The two CI failures don't seem to relate to this branch, I've pinged the backend CI channel about them. Curiously they passed in master on Thursday, failed on our branch on Friday, and fail on master for me locally today.

**Comment by crisptrutski** [comment]
Decided to wait on 51 release branch getting cut before merging this.

**Comment by crisptrutski** [comment]
51 is cut, tally ho!

**Pull request merged**
